tpm2: Add NVReadLock and NVWriteLock functions

pschultz opened the pull request:
Add functions for the TPM2_NV_ReadLock and TPM2_NV_WriteLock commands.

The commands are specified in ["Trusted Platform Module Library
Part 3: Commands" (PDF)](https://www.trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-3-Commands-01.38.pdf), pages 391f. and 400f.; and the command codes in [Part 2 (PDF)](https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf), pages 46f.

pschultz commented:
@googlebot I signed it!

twitchy-jsonp commented on tpm2/test/tpm2_test.go:
```diff
@@ -997,6 +997,24 @@ func TestNVReadWrite(t *testing.T) {
+	err := NVWrite(rw, HandleOwner, idx, emptyPassword, data, 0)
+	switch err := err.(type) {
+	case nil:
+		t.Fatalf("NVWrite succeeded after NVWriteLock")
```
Use `t.Fatal` here as you do not have any additional values aside from the message string.

twitchy-jsonp commented on tpm2/test/tpm2_test.go:
```diff
@@ -1014,6 +1032,19 @@ func TestNVReadWrite(t *testing.T) {
+	}
+
+	// Read the data again. Should fail now because it's read-locked.
+	_, err = NVReadEx(rw, idx, HandleOwner, emptyPassword, 0)
```
These can be rolled into one line:

```
if _, err = NVReadEx(rw, idx, HandleOwner, emptyPassword, 0); err != nil {
  ...
}
```

twitchy-jsonp commented on tpm2/test/tpm2_test.go:
```diff
@@ -1014,6 +1032,19 @@ func TestNVReadWrite(t *testing.T) {
+	// Read the data again. Should fail now because it's read-locked.
+	_, err = NVReadEx(rw, idx, HandleOwner, emptyPassword, 0)
+	if err == nil {
+		t.Fatalf("NVRead succeeded after NVReadLock")
```
Use `t.Fatal` here

pschultz commented:
> For the test (`TestNVReadWrite`), consider splitting it up so the NV locking is tested in a different function (preferable), or update the name of the test method.

I started out with a separate function, but it was pretty much a carbon copy of the existing test. I changed the name in the amended commit and addressed the other comments too.
